app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,133 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val popularCategories = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES
+    )
+
+    private val nowPlayingCategories = listOf(
+        MovieCategory.NOW_PLAYING_MOVIES,
+        MovieCategory.NOW_PLAYING_TV
+    )
+
+    private val upcomingCategories = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK
+    )
+
+    private val popularSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.POPULAR_STREAMING)
+    private val upcomingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.UPCOMING_TODAY)
+    private val nowPlayingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.NOW_PLAYING_TV)
+
+    private val _popularMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val popularMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularMoviesHomeViewState.asStateFlow()
+
+    private val _nowPlayingMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val nowPlayingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _nowPlayingMoviesHomeViewState.asStateFlow()
+
+    private val _upcomingMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val upcomingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _upcomingMoviesHomeViewState.asStateFlow()
+
+    init {
+        getPopularMovies()
+        getUpcomingMovies()
+        getNowPlayingMovies()
+    }
+
+    private fun getPopularMovies() {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                _popularMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategories,
+                    selectedMovieCategory = popularSelectedCategory.value,
+                    movies = it
+                )
+            }
+        }
```

Comment:
Instead of collecting and then changing ViewState we can just map the flow. So the idea is that you always transform the flow using one of many extension functions it offers, and then when you need it (in View), you collect it. 
Also right now you are not observing category flow, you are just taking the latest value once the popularMovies emit something. Your application works correctly ebcause you are emitting new value to ViewState, with `get__Movies` method every time you change the category and that's why you read the latest value. But this is not how we want to implement this, you should observe both repo and selected category and then create a ViewState based on it, and when any of those changes your viewState will be emitted with new value. 
So the idea is to do something like this: 
`selected.flatMapLatest { repo.movies().map{ ...}}.stateIn(...)`